chain/client/src/tx_request_handler.rs

```diff
@@ -0,0 +1,420 @@
+    actix::SyncArbiter::start(config.transaction_request_handler_threads, move || {
+        let view_client_actor = TxRequestHandler::new(
+            clock.clone(),
+            config.clone(),
+            tx_pool.clone(),
+            epoch_manager.clone(),
+            shard_tracker.clone(),
+            validator_signer.clone(),
+            runtime.clone(),
+            chain_genesis.clone(),
+            network_adapter.clone(),
+        )
+        .unwrap();
+        SyncActixWrapper::new(view_client_actor)
+    })
+}
+
+#[derive(Clone)]
+struct TxRequestHandlerConfig {
+    tx_routing_height_horizon: u64,
+    epoch_length: u64,
+}
+
+/// Accepts `process_tx` requests. Pushes the incoming transactions to the pool.
+pub struct TxRequestHandler {
+    config: TxRequestHandlerConfig,
+    tx_pool: Arc<Mutex<ShardedTransactionPool>>,
+    chain: Chain,
+    epoch_manager: Arc<dyn EpochManagerAdapter>,
+    shard_tracker: ShardTracker,
+    validator_signer: MutableValidatorSigner,
+    runtime: Arc<dyn RuntimeAdapter>,
+    network_adapter: PeerManagerAdapter,
+}
+
+impl TxRequestHandler {
+    pub fn new(
+        clock: Clock,
+        config: ClientConfig,
+        tx_pool: Arc<Mutex<ShardedTransactionPool>>,
+        epoch_manager: Arc<dyn EpochManagerAdapter>,
+        shard_tracker: ShardTracker,
+        validator_signer: MutableValidatorSigner,
+        runtime: Arc<dyn RuntimeAdapter>,
+        chain_genesis: ChainGenesis,
+        network_adapter: PeerManagerAdapter,
+    ) -> Result<Self, near_client_primitives::types::Error> {
+        let my_config = TxRequestHandlerConfig {
+            tx_routing_height_horizon: config.tx_routing_height_horizon,
+            epoch_length: config.epoch_length,
+        };
+
+        let chain = Chain::new_for_view_client(
+            clock.clone(),
+            epoch_manager.clone(),
+            shard_tracker.clone(),
+            runtime.clone(),
+            &chain_genesis,
+            near_chain::DoomslugThresholdMode::TwoThirds,
+            config.save_trie_changes,
+        )?;
+
+        Ok(Self {
+            config: my_config,
+            tx_pool,
+            validator_signer,
+            chain,
+            epoch_manager,
+            runtime,
+            shard_tracker,
+            network_adapter,
+        })
+    }
+
+    /// Submits the transaction for future inclusion into the chain.
+    ///
+    /// If accepted, it will be added to the transaction pool and possibly forwarded to another validator.
+    #[must_use]
+    pub fn process_tx(
+        &mut self,
+        tx: SignedTransaction,
+        is_forwarded: bool,
+        check_only: bool,
+    ) -> ProcessTxResponse {
+        let signer = self.validator_signer.get();
+        unwrap_or_return!(self.process_tx_internal(&tx, is_forwarded, check_only, &signer), {
+            let me = signer.as_ref().map(|signer| signer.validator_id());
+            tracing::warn!(target: "client", ?me, ?tx, "Dropping tx");
```

Comment:
Nit: node operators are pretty trigger happy about warnings showing up in their logs, so unless a warning represents something truly outstanding (which this probably isn't -- it seems like fairly normal for a transaction to be dropped for many reasons) this should be a debug or lower.